client/homebrew/pages/editPage/editPage.jsx

```diff
@@ -52,35 +57,81 @@ const EditPage = createClass({
 
 	getInitialState : function() {
 		return {
-			brew                   : this.props.brew,
+			brew : {
+				text        : this.props.brew.text || '',
+				style       : this.props.brew.style || '',
+				shareId     : this.props.brew.shareId || null,
+				editId      : this.props.brew.editId || null,
+				gDrive      : this.props.brew.gDrive || false,
+				title       : this.props.brew.title || '',
+				description : this.props.brew.description || '',
+				tags        : this.props.brew.tags || '',
+				published   : this.props.brew.published || false,
+				authors     : this.props.brew.authors || [],
+				systems     : this.props.brew.systems || [],
+				renderer    : this.props.brew.renderer || 'legacy'
+			},
 			isSaving               : false,
 			isPending              : false,
 			alertTrashedGoogleBrew : this.props.brew.trashed,
 			alertLoginToTransfer   : false,
-			saveGoogle             : this.props.brew.googleId ? true : false,
+			saveGoogle             : (global.account && global.account.googleId ? true : false),
 			confirmGoogleTransfer  : false,
 			errors                 : null,
 			htmlErrors             : Markdown.validate(this.props.brew.text),
-			url                    : ''
+			url                    : '',
+			editType               : ''
 		};
 	},
 	savedBrew : null,
 
 	componentDidMount : function(){
-		this.setState({
-			url : window.location.href
+		// Determine editor type
+		editTypes.forEach((type)=>{
+			if(this.props.editType == type) {
+				this.editType = this.props.editType;
+			};
 		});
+		if(!this.editType){
+			// ERROR: Unknown editType
+			console.log('Unknown EditType parameter!');
+			// Redirect to Home page; need to revisit with better Error messaging in the future
+			window.location.href = `/`;
+		};
+
+		if(this.editTypeIsEdit()) {
+			this.saveGoogle = this.props.brew.googleId ? true : false;
+		}
 
 		this.savedBrew = JSON.parse(JSON.stringify(this.props.brew)); //Deep copy
 
-		this.trySave();
+		const brew = this.props.brew;
+
+		if(this.editTypeIsNew()) {
+			console.log('is new');
+			if((!brew.text && localStorage.getItem(BREWKEY)) && (!brew.style && localStorage.getItem(STYLEKEY))){
+				console.log('has local data');
+
+				const brewStorage  = localStorage.getItem(BREWKEY) ?? '';
+				const styleStorage  = localStorage.getItem(STYLEKEY) ?? '';
+
+				brew.text = brew.text || (brewStorage  ?? '');
+				brew.style = brew.style || (styleStorage ?? '');
+			}
+		}
+
+		if(this.editTypeIsEdit()) { this.trySave(); };
 		window.onbeforeunload = ()=>{
-			if(this.state.isSaving || this.state.isPending){
+			if(!editTypeIsNew && (this.state.isSaving || this.state.isPending)){
 				return 'You have unsaved changes!';
 			}
 		};
 
+		console.log(brew);
+
 		this.setState((prevState)=>({
+			brew       : _.merge({}, prevState.brew, brew),
```

Comment:
Just FYI: You can have separate `this.setState` calls; you don't have to bunch all of the variables together in one call like this if it makes the code easier to decipher. The only issue is if you have two or more `setState` that modify the same variable, then you might have a conflict.

Reply:
As I recall, I was getting errors where `this.state.brew.text` was still undefined after `setState` had been called to set it. I put it down to the asynchronous nature of `setState` and reduced the number of calls by creating a `const brew = this.state.brew`, applying the changes to that object, then merging it in a single `setState` at the end of the function. I stopped getting the error, so I moved on.

Reply:
I see. Yes, that is another thing to think about; `setState` doesn't immediately update, so you can't start querying it's values like, in the same function. If this is working for you now then you can keep it as is.